docs/src/tutorials/linear/diet.jl

```diff
@@ -90,32 +90,27 @@ limits = DataFrames.DataFrame(
 # HiGHS as our optimizer:
 
 model = Model(HiGHS.Optimizer)
+set_silent(model)
 
-# Next, we create a set of decision variables `x`, indexed over the foods in the
-# `data` DataFrame. Each `x` has a lower bound of `0`.
+# Next, we create a set of decision variables `x`, with one element for each row
+# in the DataFrame. Each `x` has a lower bound of `0`, and we store the vector
+# as a new column `x` in the DataFrame `foods`:
 
-@variable(model, x[foods.name] >= 0);
+@variable(model, x[foods.name] >= 0)
+foods.x = Array(x)
```

Comment:
@jd-foster: I changed it to this so that `x` has meaningful indices instead of `x[1]` through x[n]`. Too magical?

Reply:
I thought about this change, and concluded it didn't matter if the index matched the name since it's row already has the key column. It matters if you split the JuMP variable column off the dataframe.